### PR TITLE
feat: allow to pass extra data to upstreams endpoints

### LINKED ISSUE #469: Support passing extra data to the model's upstreams
Occasionally for providing a sophisticated load balancing it's essential to pass an extra metadata to the upstreams, so the destination/adapter will have a better context of how to process the request. The idea is to extend the current configuration for the upstreams by introducing a new optional block `extraData` which will be passed to a particular upstream endpoint as header `X-UPSTREAM-EXTRA-DATA`. 

Example of the configuration:
```json
{
  "models": {
    "llm": {
      "type": "chat",
      "upstreams": [
        {
          "extraData": {
            "<key-1>":"<value-1>",
            "<key-2>":"<value-2>"
          }
        },
        {
          "extraData": {
            "<key-3>":"<value-3>",
            "<key-4>":"<value-4>"
          }
        }
      ]
    }
  }
}
```

### PR DESCRIPTION
**Pull request by alekseyvdovenko**
Occasionally for providing a sophisticated load balancing it's essential to pass an extra metadata to the upstreams, so the destination/adapter will have a better context of how to process the request. The idea is to extend the current configuration for the upstreams by introducing a new optional block `extraData` which will be passed to a particular upstream endpoint as header `X-UPSTREAM-EXTRA-DATA`. 

Example of the configuration:
```json
{
  "models": {
    "llm": {
      "type": "chat",
      "upstreams": [
        {
          "extraData": {
            "<key-1>":"<value-1>",
            "<key-2>":"<value-2>"
          }
        },
        {
          "extraData": {
            "<key-3>":"<value-3>",
            "<key-4>":"<value-4>"
          }
        }
      ]
    }
  }
}
```

### CODE REVIEW

**Comment by alekseyvdovenko** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-core-pr-470.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by alekseyvdovenko** [comment]
This is the blocker for https://github.com/epam/ai-dial-adapter-bedrock/issues/126

**Comment by sdryapko** [comment]
@astsiapanay  please review

File: src/main/java/com/epam/aidial/core/util/ToStringDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package com.epam.aidial.core.util;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+
+import java.io.IOException;
+
+public class ToStringDeserializer extends JsonDeserializer<String> {
```
Comment (astsiapanay) [inline src/main/java/com/epam/aidial/core/util/ToStringDeserializer.java]: JsonToStringDeserializer?

Comment (alekseyvdovenko) [reply src/main/java/com/epam/aidial/core/util/ToStringDeserializer.java]: renamed

**Review by astsiapanay — APPROVED** [review]

**Comment by alekseyvdovenko** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-core-pr-470.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by alekseyvdovenko** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-core-pr-470.nightly-test.deltixhub.io
>E2E tests status: success

**Pull request merged**